### PR TITLE
Support no proxy

### PR DESCRIPTION
**Pull request by arightious**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -7,19 +7,19 @@ data "aws_region" "current" {}
 
 module "services_iam_role" {
   # source = "./modules/iam/services_role"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/services_role?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/services_role?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 2]
```diff
@@ -7,19 +7,19 @@ data "aws_region" "current" {}
...
 
   platform_instance_id = var.platform_instance_id
 }
 
 module "amis" {
   # source = "./modules/amis"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/amis?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/amis?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 3]
```diff
@@ -7,19 +7,19 @@ data "aws_region" "current" {}
...
 }
 
 module "network" {
   # source = "./modules/network/vpc"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/network/vpc?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/network/vpc?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 4]
```diff
@@ -43,7 +43,7 @@ module "network" {
 
 module "nsg" {
   # source = "./modules/security_groups"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/security_groups?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/security_groups?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 5]
```diff
@@ -61,14 +61,14 @@ module "nsg" {
 
 module "proxy_iam" {
   # source = "./modules/iam/proxy"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/proxy?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/proxy?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 6]
```diff
@@ -61,14 +61,14 @@ module "nsg" {
...
 
   platform_instance_id = var.platform_instance_id
 }
 
 module "proxy_loadbalancer" {
   # source = "./modules/loadbalancers/proxy"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/loadbalancers/proxy?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/loadbalancers/proxy?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 7]
```diff
@@ -79,7 +79,7 @@ module "proxy_loadbalancer" {
 
 module "proxy_asg" {
   #source = "./modules/proxy"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/proxy?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/proxy?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 8]
```diff
@@ -104,7 +104,7 @@ module "proxy_asg" {
 
 module "proxy_network" {
   # source = "./modules/network/proxy"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/network/proxy?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/network/proxy?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 9]
```diff
@@ -117,15 +117,15 @@ module "proxy_network" {
 
 module "cloudwatch_logs" {
   # source = "./modules/cloudwatch"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/cloudwatch?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/cloudwatch?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 10]
```diff
@@ -117,15 +117,15 @@ module "proxy_network" {
...
 
   log_retention        = var.log_retention
   platform_instance_id = var.platform_instance_id
 }
 
 module "dashboard" {
   #source = "./modules/dashboard"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/dashboard?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/dashboard?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 11]
```diff
@@ -139,15 +139,14 @@ module "dashboard" {
 
 module "queues" {
   # source = "./modules/queues"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/queues?ref=master"
-
+  source = "github.com/graymeta/terraform12-aws-platform//modules/queues?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 12]
```diff
@@ -139,15 +139,14 @@ module "dashboard" {
...
 
   platform_instance_id = var.platform_instance_id
 }
 
 module "elasticache" {
   # source = "./modules/elasticache"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/elasticache?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/elasticache?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 13]
```diff
@@ -157,7 +156,7 @@ module "elasticache" {
 
 module "rds" {
   # source = "./modules/rds/services"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/rds/services?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/rds/services?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 14]
```diff
@@ -178,7 +177,7 @@ module "rds" {
 
 module "elasticsearch" {
   # source = "./modules/elasticsearch"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/elasticsearch?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/elasticsearch?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 15]
```diff
@@ -195,15 +194,15 @@ module "elasticsearch" {
 
 module "ecs_iam" {
   # source = "./modules/iam/ecs"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/ecs?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/ecs?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 16]
```diff
@@ -195,15 +194,15 @@ module "elasticsearch" {
...
 
   platform_instance_id = var.platform_instance_id
   temp_bucket          = var.temp_bucket
 }
 
 module "ecs" {
   # source = "./modules/ecs"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/ecs?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/ecs?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 17]
```diff
@@ -226,7 +225,7 @@ resource "aws_sns_topic" "harvest_complete" {
 
 module "services_iam" {
   # source = "./modules/iam/services_policy"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/services_policy?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/services_policy?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 18]
```diff
@@ -255,7 +254,7 @@ module "services_iam" {
 
 module "services_alb" {
   # source = "./modules/loadbalancers/services"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/loadbalancers/services?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/loadbalancers/services?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 19]
```diff
@@ -268,7 +267,7 @@ module "services_alb" {
 
 module "services" {
   # source = "./modules/services"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/services?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/services?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline main.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 20]
```diff
@@ -383,4 +382,4 @@ module "services" {
 
 output "GrayMetaPlatformEndpoint" {
   value = module.services_alb.services_endpoint
-}
+}
```
Comment (jasonsattler) [inline main.tf]: we should have 1 newline at end of files

File: mlservices.tf [hunk 21]
```diff
@@ -1,216 +1,215 @@
...
+#   platform_instance_id            = var.platform_instance_id
+#   proxy_endpoint                  = module.proxy_loadbalancer.proxy_endpoint
+#   target_group_mlservices_arn     = module.mlservices_alb.target_group_tcues_arn
+#   user_init                       = ""
+#   volume_size                     = 50
+# }
+
+# output "mlservices_tcues" {
+#   value = module.mlservices_tcues.endpoint
+# }
```
Comment (jasonsattler) [inline mlservices.tf]: So are we changing the default deployment.  If we comment this all out then we need to make sure the variables in the main.tf are set to empty string.

Comment (jasonsattler) [reply mlservices.tf]: Also fix the source references in this file.  It should never be master or a branch.  Instead the next release tag version.

File: mlservices.tf [hunk 22]
```diff
@@ -1,216 +1,215 @@
...
 
 # ##########################################
 # # MLServices VSSOCCER
 # module "mlservices_vssoccer" {
 #   # source = "./modules/mlservices/vssoccer"
-#   source = "github.com/graymeta/terraform12-aws-platform//modules/mlservices/vssoccer?ref=master"
+#   source = "github.com/graymeta/terraform12-aws-platform//modules/mlservices/vssoccer?ref=Support-No-Proxy"
```
Comment (jasonsattler) [inline mlservices.tf]: Need to update the ref to the next release version.

File: main.tf [hunk 23]
```diff
@@ -294,6 +293,7 @@ module "services" {
   elasticsearch_endpoint            = module.elasticsearch.elasticsearch_endpoint
   encrypted_config_blob             = var.encrypted_config_blob
   encryption_key                    = var.encryption_key
+  mlservices_endpoint               = module.mlservices_alb.mlservices_endpoint
   faces_endpoint                    = coalesce(module.mlservices_faces.faces_endpoint, var.faces_endpoint)
```
Comment (jasonsattler) [inline main.tf]: if you are going to comment out MLServices by default then these two variables need to be blanked.

File: modules/mlservices/audio/main.tf [hunk 24]
```diff
@@ -7,11 +7,11 @@ locals {
...
 
-  api_port       = local.api_port
-  log_group      = "GrayMetaPlatform-${var.platform_instance_id}-ML"
+  api_port     = local.api_port
+  log_group    = "GrayMetaPlatform-${var.platform_instance_id}-ML"
   proxy_endpoint = var.proxy_endpoint
-  service_name   = local.api_name
-  tfs_port       = local.tfs_port
+  service_name = local.api_name
+  tfs_port     = local.tfs_port
```
Comment (jasonsattler) [inline modules/mlservices/audio/main.tf]: Run fmt on this file

File: modules/mlservices/ml_init/main.tf [hunk 25]
```diff
@@ -5,10 +5,10 @@ data "template_file" "userdata" {
...
   vars = {
-    api_port       = var.api_port
-    log_group      = var.log_group
+    api_port     = var.api_port
+    log_group    = var.log_group
     proxy_endpoint = var.proxy_endpoint
-    service_name   = var.service_name
-    tfs_port       = var.tfs_port
+    service_name = var.service_name
+    tfs_port     = var.tfs_port
```
Comment (jasonsattler) [inline modules/mlservices/ml_init/main.tf]: Run fmt on this

File: modules/mlservices/ml_init/main.tf [hunk 26]
```diff
@@ -5,10 +5,10 @@ data "template_file" "userdata" {
   template = file("${path.module}/userdata.tpl")
```
Comment (jasonsattler) [inline modules/mlservices/ml_init/main.tf]: Should we be switching this to sh script with the proxy check?

File: modules/mlservices/faces/userdata.tpl [hunk 27]
```diff
@@ -179,4 +177,4 @@ write_files:
         HTTP_PROXY=http://${proxy_endpoint}
         HTTPS_PROXY=http://${proxy_endpoint}
         NO_PROXY=169.254.169.254
-    path: /var/awslogs/etc/proxy.conf
+    path: /var/awslogs/etc/proxy.conf
```
Comment (jasonsattler) [inline modules/mlservices/faces/userdata.tpl]: Should we be switching this to sh script with the proxy check?

File: modules/mlservices/ml_init/userdata.tpl [hunk 28]
```diff
@@ -9,14 +9,12 @@ runcmd:
...
 - growpart /dev/xvda 2
 - growpart /dev/nvme0n1 2
 - pvresize /dev/xvda2
 - pvresize /dev/nvme0n1p2
 - lvextend -l +100%FREE /dev/mapper/centos-root
 - xfs_growfs /dev/mapper/centos-root
-- yum install chrony -y
 - systemctl enable chronyd
 - systemctl start chronyd
 - systemctl daemon-reload
```
Comment (jasonsattler) [inline modules/mlservices/ml_init/userdata.tpl]: Should we be switching this to sh script with the proxy check?

File: modules/mlservices/nld/main.tf [hunk 29]
```diff
@@ -7,11 +7,11 @@ locals {
...
+  api_port     = local.api_port
+  log_group    = "GrayMetaPlatform-${var.platform_instance_id}-ML"
   proxy_endpoint = var.proxy_endpoint
-  service_name   = local.api_name
-  tfs_port       = local.tfs_port
+  service_name = local.api_name
+  tfs_port     = local.tfs_port
 }
 
 # Create the cluster
```
Comment (jasonsattler) [inline modules/mlservices/nld/main.tf]: run fmt on this

File: modules/mlservices/nlp/main.tf [hunk 30]
```diff
@@ -7,10 +7,10 @@ data "template_file" "userdata" {
...
   vars = {
-    api_port       = local.api_port
-    log_group      = "GrayMetaPlatform-${var.platform_instance_id}-ML"
+    api_port     = local.api_port
+    log_group    = "GrayMetaPlatform-${var.platform_instance_id}-ML"
     proxy_endpoint = var.proxy_endpoint
-    service_name   = local.api_name
+    service_name = local.api_name
   }
 }
```
Comment (jasonsattler) [inline modules/mlservices/nlp/main.tf]: Should we be switching this to sh script with the proxy check?

File: modules/mlservices/nlp/userdata.tpl [hunk 31]
```diff
@@ -62,13 +60,13 @@ write_files:
...
--   content: |
-        HTTP_PROXY=http://${proxy_endpoint}
-        HTTPS_PROXY=http://${proxy_endpoint}
-        NO_PROXY=169.254.169.254
-    path: /var/awslogs/etc/proxy.conf
+-   content: |	
+        HTTP_PROXY=http://${proxy_endpoint}	
+        HTTPS_PROXY=http://${proxy_endpoint}	
+        NO_PROXY=169.254.169.254	
+    path: /var/awslogs/etc/proxy.conf
```
Comment (jasonsattler) [inline modules/mlservices/nlp/userdata.tpl:72]: Should we be switching this to sh script with the proxy check?

File: modules/mlservices/object/main.tf [hunk 32]
```diff
@@ -7,11 +7,11 @@ locals {
...
-  api_port       = local.api_port
-  log_group      = "GrayMetaPlatform-${var.platform_instance_id}-ML"
+  api_port     = local.api_port
+  log_group    = "GrayMetaPlatform-${var.platform_instance_id}-ML"
   proxy_endpoint = var.proxy_endpoint
-  service_name   = local.api_name
-  tfs_port       = local.tfs_port
+  service_name = local.api_name
+  tfs_port     = local.tfs_port
 }
```
Comment (jasonsattler) [inline modules/mlservices/object/main.tf]: run fmt

File: modules/mlservices/tcues/main.tf [hunk 33]
```diff
@@ -7,11 +7,11 @@ locals {
...
-  api_port       = local.api_port
-  log_group      = "GrayMetaPlatform-${var.platform_instance_id}-ML"
+  api_port     = local.api_port
+  log_group    = "GrayMetaPlatform-${var.platform_instance_id}-ML"
   proxy_endpoint = var.proxy_endpoint
-  service_name   = local.api_name
-  tfs_port       = local.tfs_port
+  service_name = local.api_name
+  tfs_port     = local.tfs_port
 }
```
Comment (jasonsattler) [inline modules/mlservices/tcues/main.tf]: run fmt

File: modules/mlservices/vssoccer/main.tf [hunk 34]
```diff
@@ -7,11 +7,11 @@ locals {
...
-  api_port       = local.api_port
-  log_group      = "GrayMetaPlatform-${var.platform_instance_id}-ML"
+  api_port     = local.api_port
+  log_group    = "GrayMetaPlatform-${var.platform_instance_id}-ML"
   proxy_endpoint = var.proxy_endpoint
-  service_name   = local.api_name
-  tfs_port       = local.tfs_port
+  service_name = local.api_name
+  tfs_port     = local.tfs_port
 }
```
Comment (jasonsattler) [inline modules/mlservices/vssoccer/main.tf]: run fmt

**Review by jasonsattler — CHANGES_REQUESTED** [review]
Wondering if we don't move the noproxy directory over to examples.  I think we kinda don't want people to use that as a default.

File: modules/mlservices/nlp/userdata.tpl (hunk 31 above)
Comment (arightious) [reply modules/mlservices/nlp/userdata.tpl:72]: changed all the ML, and the proxy module to sh

File: mlservices.tf (hunk 21 above)
Comment (arightious) [reply mlservices.tf]: done, and updated the docs to reflect this.

File: docs/encryption.md [hunk 35]
```diff
@@ -24,7 +24,7 @@ data "aws_region" "current" {}
 
 module "services_iam_role" {
   # source = "./modules/iam/services_role"
-  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/services_role?ref=master"
+  source = "github.com/graymeta/terraform12-aws-platform//modules/iam/services_role?ref={tag}"
```
Comment (jasonsattler) [inline docs/encryption.md:27]: v0.0.2

**Pull request closed without merging**